src/test/java/cn/leancloud/leanengine_test/IMHookIntegrationTest.java

```diff
@@ -1,24 +1,53 @@
 package cn.leancloud.leanengine_test;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Map;
 
-import junit.framework.TestCase;
-
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.Test;
 
 import com.avos.avoscloud.AVCloud;
 import com.avos.avoscloud.AVException;
 import com.avos.avoscloud.AVOSCloud;
+import com.avos.avoscloud.internal.impl.EngineRequestSign;
+
+import cn.leancloud.LeanEngine;
+
+public class IMHookIntegrationTest {
+
+  private static Server server;
+  private static int port = 3000;
 
-public class IMHookIntegrationTest extends TestCase {
-  @Override
-  public void setUp() {
+  @BeforeClass
+  public static void setUp() throws Exception {
```

Comment:
可以的，我当时想弄，但是 beforeClass 是一个 static 方法，所以暂时放弃了，可以抽取下。

lbt05 <notifications@github.com>于2016年7月14日周四 上午10:46写道：

> In src/test/java/cn/leancloud/leanengine_test/IMHookIntegrationTest.java
> <https://github.com/lbt05/leanengine-java-sdk/pull/2#discussion_r70741539>
> :
>
> >
> > -public class IMHookIntegrationTest extends TestCase {
> > -  @Override
> > -  public void setUp() {
> > +  @BeforeClass
> > +  public static void setUp() throws Exception {
>
> 要不要搞一个EngineBasicTest来写这个代码。。感觉反复写很蛋疼啊
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lbt05/leanengine-java-sdk/pull/2/files/a7c641929a06bd549ae42b44d951f1bf2c973a13#r70741539>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABE2MH1mj0KU83Pi8bD0bPHAG9r8MlUnks5qVaMDgaJpZM4JKR_o>
> .
>